### ISSUE TITLE
ANTLR 4.12.0 may produce Java sources with illegal unicode escapes under Windows if a package name begins with 'u'

### ISSUE THREAD

**Issue by DanySK** [antlr/antlr4#4128]
This is a regression, as 4.11.1 works correctly.

For files with a package starting in `u`, for instance `it.unibo.xxx`, when used under Windows, ANTLR 4.12.0 produces a header comment in the form:

```
// Generated from it\unibo\alchemist\model\internal\biochemistry\dsl\Biochemistrydsl.g4 by ANTLR 4.12.0
```

This is problematic, however, as the Java compiler reports it as `error: illegal unicode escape`

Here is a gradle scan showing the issue:
https://scans.gradle.com/s/uxdeejknwdmve/console-log?task=:alchemist-incarnation-biochemistry:compileJava

Quick reproduction can be achieved with the following:
```
git clone https://github.com/AlchemistSimulator/Alchemist.git
cd Alchemist
git checkout renovate/antlr4
./gradlew :alchemist-incarnation-biochemistry:compileJava
```

The fix should be reasonably easy; I'd suggest converting backslashes to slashes in the header (also, this would generate the same comment code for all the platforms).

**Comment by DanySK** [comment]
I believe [this line](https://github.com/antlr/antlr4/blob/1f9a474cc2dca0ec2ab3ce42d2dc2468205647c7/tool/resources/org/antlr/v4/tool/templates/codegen/Java/Java.stg#L220) should be the culprit.

**Comment by sschuberth** [comment]
For reference, this seems to be a regression of https://github.com/antlr/antlr4/issues/164.

**Comment by DanySK** [comment]
Yes, it is.

**Comment by DanySK** [comment]
The issue still stands in 4.13.0. Is this on the radar for a solution? Otherwise, I might decide to work it around by running an additional code generation task that modifies ANTLR4-generated sources on Windows.

**Comment by sschuberth** [comment]
> Otherwise, I might decide to work it around by running an additional code generation task that modifies ANTLR4-generated sources on Windows.

That's what I did as well now:

```
tasks.withType<AntlrTask>().configureEach {
    doLast {
        // Work around https://github.com/antlr/antlr4/issues/4128.
        outputDirectory.walk()
            .filter { it.isFile && it.extension == "java" }
            .forEach { javaFile ->
                val lines = javaFile.readLines()

                val text = buildString {
                    lines.mapIndexed { index, line ->
                        val patchedLine = when {
                            index == 0 && line.startsWith("// Generated from ") -> line.replace('\\', '/')
                            else -> line
                        }

                        appendLine(patchedLine)
                    }
                }

                javaFile.writeText(text)
            }
    }
}
```

**Comment by jimidle** [comment]
I think that the correct answer here is not use the Windows `\` but change that to use '/', which is in fact supported by Windows anyway in most cases, and anyway, this is just documentation in the generated code. 

If you guys agree, I will go take a look at changing that, or if that is a pain, at least using \\

**Comment by jimidle** [comment]
The PR that is waiting against dev fixes this issue, but it is not included in the 4.13.0 release unfortunately.

**Comment by KvanTTT** [comment]
@parrt it can be closed.
